File: _Logic/JDBC/src/main/java/app/booking/db/Booking.java
package app.booking.db;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

import java.sql.Date;
import java.sql.Time;
import java.sql.Timestamp;

//@AllArgsConstructor
//@Getter
//@Setter
public class Booking {
    private Integer bookingId;
    private Timestamp timestamp;
    private Date date;
    private Time start;
    private Time end;
    private String cityId;
    private String centerId;
    private String courtId;
    private String playerId;
    private Integer status;

    public Booking(Integer bookingId, Timestamp timestamp, Date date, Time start, Time end, String cityId, String centerId, String courtId, String playerId, Integer status) {
        this.bookingId = bookingId;
        this.timestamp = timestamp;
        this.date = date;
        this.start = start;
        this.end = end;
        this.cityId = cityId;
        this.centerId = centerId;
        this.courtId = courtId;
        this.playerId = playerId;
        this.status = status;
    }

    public Integer getBookingId() {
        return bookingId;
    }

    public Timestamp getTimestamp() {
        return timestamp;
    }

    public Date getDate() {
        return date;
    }

    public Time getStart() {
        return start;
    }

    public Time getEnd() {
        return end;
    }

    public String getCityId() {
        return cityId;
    }

    public String getCenterId() {
        return centerId;
    }

    public String getCourtId() {
        return courtId;
    }

    public String getPlayerId() {
        return playerId;
    }

    public Integer getStatus() {
        return status;
    }
}

File: _Presentation/CourtBooking/app/src/main/java/com/example/courtbooking/adapter/Booking.kt
package com.example.courtbooking.adapter

import org.json.JSONObject
import java.sql.Date
import java.sql.Timestamp

class Booking {
    private var bookingId:String
    private var cityId:String
    private var centerId:String
    private var courtId:String
    private var date: String
    private var timestamp: String
    private var start:String
    private var end:String
    private var playerId:String
    private var status: Int
    private var cardId: Int

    constructor(booking: JSONObject) {
        bookingId = booking.getInt("bookingId").toString()
        cityId = booking.getString("cityId")
        centerId = booking.getString("centerId")
        courtId = booking.getString("courtId")
        date = Date(booking.getLong("date")).toString()
        timestamp = Timestamp(booking.getLong("timestamp")).toString()
        start = booking.getString("start")
        end = booking.getString("end")
        playerId = booking.getString("playerId")
        status = booking.getInt("status")
        cardId = booking.getInt("cardId")
    }

    fun getBookingId(): String {
        return bookingId
    }
    fun getCityId(): String {
        return cityId
    }
    fun getCenterId(): String {
        return centerId
    }
    fun getCourtId(): String {
        return courtId
    }
    fun getDate(): String {
        return date
    }
    fun getTimestamp(): String {
        return timestamp
    }
    fun getStart(): String {
        return start
    }
    fun getEnd(): String {
        return end
    }
    fun getPlayerId(): String {
        return playerId
    }
    fun getStatus(): Int {
        return status
    }
    fun getCard(): Int {
        return cardId
    }
}